pkg/plugin/canary/fail.go

```diff
@@ -133,16 +133,19 @@ func (o *failOptions) run() error {
 	}
 
 	if eds.Spec.Strategy.Canary == nil {
-		return fmt.Errorf("the ExtendedDaemonset does not have a canary")
+		return fmt.Errorf("the ExtendedDaemonset does not have a canary strategy")
+	}
+	if eds.Status.Canary == nil {
+		return fmt.Errorf("the ExtendedDaemonset does not have an active canary deployment")
 	}
 
 	newEds := eds.DeepCopy()
 	if newEds.Annotations == nil {
 		newEds.Annotations = make(map[string]string)
 	} else if isFailed, ok := newEds.Annotations[v1alpha1.ExtendedDaemonSetCanaryFailedAnnotationKey]; ok {
-		if o.failStatus && isFailed == "true" {
+		if o.failStatus && isFailed == "true" { //nolint:goconst
```

Comment:
maybe works to create a `ValueStringTrue = "true"` and `ValueStringFalse = false` in this file: https://github.com/DataDog/extendeddaemonset/blob/master/api/v1alpha1/const.go

Reply:
or maybe move the `stringTrue` and `stringFalse` from `pkg/plugin/canary/pause.go` to `api/v1alpha1/const.go`